src/wifi.rs

```diff
@@ -1396,6 +1391,34 @@ impl<'d> Wifi for WifiDriver<'d> {
     }
 }
 
+pub struct WifiFrame {
+    buf: *mut u8,
+    len: u16,
+    eb: *mut ffi::c_void,
+}
+
+unsafe impl Send for WifiFrame {}
+
+impl WifiFrame {
+    const unsafe fn new(buf: *mut u8, len: u16, eb: *mut ffi::c_void) -> Self {
+        Self { buf, len, eb }
+    }
+
+    pub const fn as_slice(&self) -> &[u8] {
+        unsafe { core::slice::from_raw_parts(self.buf, self.len as _) }
+    }
+
+    pub fn as_mut_slice(&mut self) -> &mut [u8] {
```

Comment:
As per above, probably not a good idea. Also as per above, maybe implement `Deref`?